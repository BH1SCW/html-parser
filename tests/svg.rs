use html_parser::HtmlParser;
use indoc::indoc;

#[test]
fn it_can_parse_svg() {
    let svg = indoc!(
        r#"
        <svg  xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
            <rect x="10" y="10" height="100" width="100" style="stroke:#ff0000; fill: #0000ff"/>
        </svg>
    "#
    );
    assert!(HtmlParser::parse(&svg).is_ok());
}
