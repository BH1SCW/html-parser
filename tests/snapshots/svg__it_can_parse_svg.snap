---
source: tests/svg.rs
expression: ast
---
{
  "treeType": "documentFragment",
  "nodes": [
    {
      "element": {
        "id": null,
        "name": "svg",
        "variant": "normal",
        "attributes": {
          "xmlns": "http://www.w3.org/2000/svg",
          "xmlns:xlink": "http://www.w3.org/1999/xlink"
        },
        "classes": [],
        "nodes": [
          {
            "element": {
              "id": null,
              "name": "rect",
              "variant": "void",
              "attributes": {
                "height": "100",
                "style": "stroke:#ff0000; fill: #0000ff",
                "width": "100",
                "x": "10",
                "y": "10"
              },
              "classes": [],
              "nodes": []
            }
          }
        ]
      }
    }
  ]
}
